net/service/handlers.go

```diff
@@ -1485,9 +1567,95 @@ func (service *OpenBazaarService) handleStore(pid peer.ID, pmes *pb.Message, opt
 	return m, nil
 }
 
+func (service *OpenBazaarService) handleOrderPayment(peer peer.ID, pmes *pb.Message, options interface{}) (*pb.Message, error) {
+	// Unmarshall
+	if pmes.Payload == nil {
+		return nil, errors.New("payload is nil")
+	}
+	paymentDetails := new(pb.OrderPaymentTxn)
+	err := ptypes.UnmarshalAny(pmes.Payload, paymentDetails)
+	if err != nil {
+		return nil, err
+	}
+
+	wal, err := service.node.Multiwallet.WalletForCurrencyCode(paymentDetails.GetCoin())
+	if err != nil {
+		return nil, err
+	}
+
+	chash, err := chainhash.NewHashFromStr(paymentDetails.GetTransactionID())
+	if err != nil {
+		return nil, err
+	}
+
+	txn, err := wal.GetTransaction(*chash)
+	if err != nil {
+		return nil, err
+	}
+
+	contract, _, _, _, _, _, err := service.datastore.Sales().GetByOrderId(paymentDetails.OrderID)
+	if err != nil {
+		return nil, net.OutOfOrderMessage
```

Comment:
@placer14 - I am returning 'out of order' error because if this message 'ORDER_PAYMENT' is being processed and there is no order in the vendor database, it would imply that 'ORDER_PAYMENT' has been processed before 'ORDER'. The 'OutOfOrderMessage' would ensure that it is re-processed in the correct order after all the offline messages are downloaded by the vendor node.

Reply:
Good point. Thank you.